pwndbg/emu/emulator.py

```diff
@@ -220,12 +220,28 @@ def __init__(self) -> None:
         # (address_successfully_executed, size_of_instruction)
         self.last_single_step_result = InstructionExecutedResult(None, None)
 
-        # Initialize the register state
-        for reg in (
-            list(self.regs.retaddr)
+        # The specific order of this list is very important:
+        # Due to the behavior of Arm in the Unicorn engine,
+        # we must write the flags register after PC, and the stack pointer after the flags register.
+        # Otherwise, the values will be clobbered
+        # https://github.com/pwndbg/pwndbg/pull/2337
+        raw_list = (
+            [self.regs.pc]
+            + list(self.regs.flags)
+            + [self.regs.stack, self.regs.frame]
+            + list(self.regs.retaddr)
             + list(self.regs.misc)
-            + list(self.regs.common)  # this includes the flags register
-        ):
+            + list(self.regs.gpr)
+        )
```

Comment:
I don't remember but how often is this called? If often, then since this order is always the same, shall we maybe have a `regs.emulator_regs_order` field in each regs struct instead of re-computing this each time here?